linchpin.py

```diff
@@ -28,13 +100,81 @@ def copy_files(path, dir_list, config):
+    """ Invokes linchpin playbook """
+    playbook_path = config.clipath+"/provision/"+PLAYBOOKS[playbook]
+    inventory = Inventory(loader=config.loader, variable_manager=config.variable_manager,  host_list=[])
+    config.variable_manager.extra_vars = e_vars 
+    passwords = {}
+    utils.VERBOSITY = 4
+    pbex = PlaybookExecutor(playbooks=[playbook_path], inventory=inventory, variable_manager=config.variable_manager, loader=config.loader, options=config.options, passwords=passwords)
+    if console == False:
+        cb = PlaybookCallback()
+        pbex._tqm._stdout_callback = cb
+        return_code = pbex.run()
+        results = cb.results
+    else:
+       results = pbex.run()
+    return results
+
+def search_path(name, path):
+    """ searches for files by name in a given path """
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            return os.path.join(root, name)
+
+def get_evars(lpf):
+    """ creates a group of extra vars on basis on linchpin file """
+    e_vars = []
+    for group in lpf:
+        topology = lpf[group].get("topology") 
+        layout = lpf[group].get("layout")
+        e_var_grp = {}
+        e_var_grp["topology"] = search_path(topology, os.getcwd())
+        e_var_grp["layout"] = search_path(layout, os.getcwd() )
+        if None in  e_var_grp.values():
+            display("ERROR:003")
+        e_vars.append(e_var_grp)
+    return e_vars
+        
+
 class Config(object):
+    """ Global config object accesible by all the click modules """
     def __init__(self):
         self.verbose = False
         self.env = Environment(loader=PackageLoader('linchpin', 'templates'))
         self.linchpinfile = self.env.get_template('linchpin.lpf.j2')
         self.clipath = os.path.dirname(os.path.realpath(__file__))
         self.dir_list = ["layouts","topologies"]
+        self.variable_manager = VariableManager()
+        self.loader = DataLoader()
+        self.inventory = Inventory(loader=self.loader, variable_manager=self.variable_manager,  host_list=['localhost'])
+        utils.VERBOSITY = 4
+        Options = namedtuple('Options', ['listtags', 'listtasks', 'listhosts', 'syntax', 'connection','module_path', 'forks', 'remote_user', 'private_key_file', 'ssh_common_args', 'ssh_extra_args', 'sftp_extra_args', 'scp_extra_args', 'become', 'become_method', 'become_user', 'verbosity', 'check'])
+        self.playbook_path = 'playbooks/test_playbook.yml'
+        self.options = Options(listtags=False, listtasks=False, listhosts=False, syntax=False, connection='ssh', module_path=None, forks=100, remote_user='test', private_key_file=None, ssh_common_args=None, ssh_extra_args=None, sftp_extra_args=None, scp_extra_args=None, become=True, become_method=None, become_user='root', verbosity=utils.VERBOSITY, check=False)
+        #variable_manager.extra_vars = {'hosts': 'mywebserver'} # This can accomodate various other command line arguments.`
```

Comment:
Will be changing the script according to PEP8.
Here on before PR is raised, will run through PEP8 checks